### PR TITLE
s2575 - ta13061 - single threaded FileSync version

### PR DESCRIPTION
**Pull request by hjeeves**

### CODE REVIEW

File: critwall/src/main/java/org/opencadc/critwall/FileSync.java [hunk 1]
```diff
@@ -67,49 +67,48 @@
...
     private static final Logger log = Logger.getLogger(FileSync.class);
 
     private static final int MAX_THREADS = 16;
     
     private final ArtifactDAO artifactDAO;
-    private StorageAdapter localStorage;
-    private final URI resourceID;
-    private final Capability locator;
+    private final ArtifactDAO jobArtifactDAO;
+    private final URI locatorResourceID;
```
Comment (pdowler) [inline critwall/src/main/java/org/opencadc/critwall/FileSync.java:92]: I noticed this rename from resourceID to locatorResourceID here and in FSJ... just to clarify? There is only one remote service (resourceID) that critwall looks up so I'm not sure it is warranted. It also means the name used in  FileSync ctor, and check (line~113) is different from the internal variable now. Admittedly, the name in the config is `locatorService` which is neither of these so a rename could be justified if the name *everywhere* was the same as the config item.

File: critwall/src/main/java/org/opencadc/critwall/FileSync.java [hunk 2]
```diff
@@ -118,53 +117,65 @@ public FileSync(Map<String,Object> daoConfig, StorageAdapter localStorage, URI r
...
-        //            if (locator == null) {
-        //                throw new IllegalArgumentException("invalid config: remote query service " + resourceID + " does not implement "
-        //                + Standards.SI_LOCATE);
-        //            }
-        //        } catch (IOException ex) {
-        //            throw new IllegalArgumentException("invalid config", ex);
-        //        }
+        // For managing the artifact iterator FileSync loops over
+        try {
+            // Make FileSync ArtifactDAO instance
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply critwall/src/main/java/org/opencadc/critwall/FileSync.java:126]: The daoConfig was set up in one place and used for artifactDAO instances. I wanted to have the jndiDataSourceNames be related (so tack 'jobs' onto the end) With only one being passed in to FileSync, the jndiDataSourceName needed to be pulled out & augmented.  That said, no problem changing it.

File: critwall/src/main/java/org/opencadc/critwall/FileSync.java [hunk 3]
```diff
@@ -118,53 +117,65 @@ public FileSync(Map<String,Object> daoConfig, StorageAdapter localStorage, URI r
...
-        while (bucketSelector.hasNext()) {
-            String nextBucket = bucketSelector.next();
-            log.info("processing bucket " + nextBucket);
-        }
-        throw new UnsupportedOperationException("TODO");
-    }
+        String currentArtifactInfo = "";
+
+        try {
+            while (bucketSelector.hasNext()) {
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply critwall/src/main/java/org/opencadc/critwall/FileSync.java:162]: done

File: critwall/src/main/java/org/opencadc/critwall/FileSync.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (hjeeves) [reply critwall/src/main/java/org/opencadc/critwall/FileSync.java:130]: jndiDataSourceName definition pushed into FilesSync.

File: critwall/src/main/java/org/opencadc/critwall/Main.java [hunk 4]
```diff
@@ -85,6 +85,7 @@
 import org.apache.log4j.Logger;
 import org.opencadc.inventory.db.SQLGenerator;
 import org.opencadc.inventory.storage.StorageAdapter;
+import org.opencadc.inventory.storage.fs.OpaqueFileSystemStorageAdapter;
```
[the remark this answers is not in the document]
Comment (hjeeves) [reply critwall/src/main/java/org/opencadc/critwall/Main.java]: done and done.

File: critwall/src/main/java/org/opencadc/critwall/FileSync.java (hunk 1 above)
Comment (hjeeves) [reply critwall/src/main/java/org/opencadc/critwall/FileSync.java]: switched names to locatorService.

**Pull request merged**